pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -1177,25 +1179,44 @@ func (p *PriorityQueue) movePodsToActiveOrBackoffQueue(logger klog.Logger, podIn
 	}
 }
 
-// getUnschedulablePodsWithMatchingAffinityTerm returns unschedulable pods which have
-// any affinity term that matches "pod".
+// getUnschedulablePodsToMoveViaPodEvent returns unschedulable pods which either of following conditions is met:
+// - have any affinity term that matches "pod".
+// - rejected by PodTopologySpread plugin.
 // NOTE: this function assumes lock has been acquired in caller.
-func (p *PriorityQueue) getUnschedulablePodsWithMatchingAffinityTerm(logger klog.Logger, pod *v1.Pod) []*framework.QueuedPodInfo {
+func (p *PriorityQueue) getUnschedulablePodsToMoveViaPodEvent(logger klog.Logger, pod *v1.Pod) []*framework.QueuedPodInfo {
 	nsLabels := interpodaffinity.GetNamespaceLabelsSnapshot(logger, pod.Namespace, p.nsLister)
 
 	var podsToMove []*framework.QueuedPodInfo
 	for _, pInfo := range p.unschedulablePods.podInfoMap {
+		if pInfo.UnschedulablePlugins.Has(podtopologyspread.Name) {
+			// This Pod may be schedulable now by this Pod event.
+			podsToMove = append(podsToMove, pInfo)
+			continue
+		}
+
```

Comment:
I put very simple mitigation here intentionally. Given `getUnschedulablePodsToMoveViaPodEvent` will be removed eventually with `AssignedPodAdded` and `AssignedPodUpdated` once QHint goes to GA, I don't want to put much filtering logic here - if I did so, it'd be something like QueueingHint in PodTopologySpread.

Reply:
did you consider checking for the namespace?

Reply:
Sorry, I missed a previous suggestion.
Yes, we can limit to Pods in the same namespace.